src/WebJobs.Extensions.DurableTask/Options/DurableTaskOptions.cs

```diff
@@ -171,6 +194,12 @@ internal void Validate()
                 throw new InvalidOperationException($"A non-empty {nameof(this.HubName)} configuration is required.");
             }
 
+            if (IsInNonProductionSlot() && this.isDefaultHubName)
+            {
+                throw new InvalidOperationException($"Task Hub name must be specified in host.json when using slots. See documentation on Task Hubs for " +
```

Comment:
NIT: neither of these strings require the "$", as they have no embedded variables.